plutus-chain-index-core/src/Plutus/ChainIndex/Effects.hs

```diff
@@ -78,7 +78,11 @@ makeEffect ''ChainIndexQueryEffect
 data ChainIndexControlEffect r where
 
     -- | Add a new block to the chain index by giving a new tip and list of tx.
-    AppendBlock :: Tip -> [ChainIndexTx] -> ChainIndexControlEffect ()
+    --
+    -- The extra `Bool` determines if we want to store this batch of transactions.
+    -- If `True` then yes, else only handle the tip and UTXOs.
+    -- This, for example, enables applications to skip unneeded pre-Alonzo transactions.
+    AppendBlock :: Tip -> [ChainIndexTx] -> Bool -> ChainIndexControlEffect ()
```

Comment:
To make a code a bit more descriptive, can we get an enum instead of the `Bool`; something like:

```
data BlockStoreOption = Save | DontSave
```

(feel free to come up with a better name!) Then it's a bit clearer at a glance what the `appendBlock ... True` calls are doing.

Reply:
Sorry I couldn't get back to this earlier. I've added 3 more commits addressing your suggestions. it does look way better going forward now, thanks!